internal/route/routes.go

```diff
@@ -73,14 +70,19 @@ func HandleIndex(w http.ResponseWriter, r *http.Request) {
 func HandleTodos(w http.ResponseWriter, r *http.Request) {
 	tpl := template.Must(template.ParseFiles("web/template/todo.html"))
 	vars := mux.Vars(r)
-	listID := vars["listId"]
+	listID, ok := vars["listId"]
+	if !ok {
+		log.Println("[error] listId not found")
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
 	list, err := model.GetTodosListByID(listID)
 	if err != nil {
 		w.WriteHeader(http.StatusNotFound)
 		return
 	}
 	err = tpl.ExecuteTemplate(w, "todo.html", struct {
-		TodoList model.TodoList
+		TodoList *model.TodoList
```

Comment:
Don't send copies! I mean it's not terrible, but it's good to get in the habit. One day you'll be confused why you're not changing a value in a struct and it'll bite you!

Example:

```
type MyStruct struct {
    Foo string
}

func mutate(s MyStruct) {
    s.Foo = "baz";
}

func main() {
    x := MyStruct{}
    x.Foo = "bar"
    mutate(x)
    fmt.Println(x.Foo)
    // prints "bar" !
}

```

Reply:
If you used `x := &MyStruct{}` and `func mutate(s *MyStruct) {` it would've worked as you expected.